crates/pallet-domains/src/lib.rs

```diff
@@ -190,6 +190,9 @@ mod pallet {
 
         /// Minimum operator stake required to become operator of a domain.
         type MinOperatorStake: Get<BalanceOf<Self>>;
+
+        /// Minimum number of blocks after which any finalized withdrawls are released to nominators.
+        type StakeWithdrawalLockingPeriod: Get<Self::BlockNumber>;
```

Comment:
No, this is per spec where once its moves to unlocking state, it is tracked with consensus chain block

Reply:
Seems it takes only `StakeEpochDuration` domain blocks (currently set to 5) to move into the unlock state, which is much smaller than `BlockTreePruningDepth` (currently set to 256).

Reply:
I believe the parameterization will be revisited once the block tree PR lands, should be a non-blocker for this PR.

Reply:
My point is not about the parameterization, `StakeEpochDuration` is expected to be small while `BlockTreePruningDepth` is expected to be large.

My concern is in the current implementation a malicious operator can submit a fraudulent ER and then deregister immediately, so they can get away from any slash successfully.

Reply:
> My concern is in the current implementation a malicious operator can submit a fraudulent ER and then deregister immediately, so they can get away from any slash successfully.

If they deregister, they are not yet unlocked and they will be slashed even in the unlocking state. Can you elaborate on your concern pls ?

Reply:
> they are not yet unlocked and they will be slashed even in the unlocking state

They can only be slashed within `StakeWithdrawalLockingPeriod` number of consensus blocks while the slash can happen within `BlockTreePruningDepth` number of domain blocks period.

The unlocking period is much smaller than the challenging period, so malicious operators can get away from slashing.

Reply:
`StakeWithdrawalLockingPeriod` will be long enough per our security model, so we should already see a challenge before the locking period is over. In the scenario when domain is stalled, and challenge did not come in before epoch was finished, then there is a possibility of this happening. There are some discussion on this [here](https://www.notion.so/Domains-v2-Specification-WIP-3fb0ec6e4d204c4881a7df50ef58da8f?d=d53a0b414c9e4aada3818d762c67feea&pvs=4#053c0c2ff0824206ab1d68197106deea). If you have concerns on this, let take the discussion there instead

Reply:
> In the scenario when domain is stalled, and challenge did not come in before epoch was finished, then there is a possibility of this happening.

Not only domain stalls, there are many scenarios that can lead to this (i.e. network delay, or too many bundles to fit in one consensus block, etc.), I think these scenarios are inevitable. When comes to security it is always better to have a guarantee instead of possibility IMO, WDYT on this @liuchengxu 

> There are some discussion on this [here](https://www.notion.so/Domains-v2-Specification-WIP-3fb0ec6e4d204c4881a7df50ef58da8f?d=d53a0b414c9e4aada3818d762c67feea&pvs=4#053c0c2ff0824206ab1d68197106deea). If you have concerns on this, let take the discussion there instead
 
Our discussion here is about using domain block or consensus block, but the linked discussion is about using block or epoch (it is also not clear if the block refers to domain block or consensus block),  I do remember we had an offline discussion about using domain block or consensus block on something and chose domain block at the end.

Reply:
Yes, we did have a discussion and concluded to use Consensus block instead.
Can you clarify @liuchengxu ?

Reply:
@NingLin-P in that same discussion Ved linked there is also "domain vs consensus block". My issue with using domain blocks is that we would need to track every domains block height in the unlocks list then. Is that a valid concern?

Reply:
> Would be great to see concrete example TBH.

In either scenario of domain stall, network delay, and too many bundles to fit in one consensus block, can cause no domain bundle included in a consensus block thus no domain block will be driven from that consensus block, so the consensus chain will progress faster than the domain chain, which means the unlocking period (using consensus block) will progress faster than the challenging period (using domain block).

The following example uses `BlockTreePruningDepth = 256`, `StakeWithdrawalLockingPeriod = 100` and `StakeEpochDuration = 5` as the current implementation:
- In consensus chain `#10` and domain chain `#10`, a malicious operator submits a fraudulent ER and a deregister request.
-  After consensus chain progress `#10 -> #12` and domain chain progress `#10 -> #12`, the domain epoch transited the deregistration and reached the unlocking state.
- The domain chain stalled at `#12`, but the consensus chain progressed as usual `#12 -> #112`, and the deregistration was unlock
- The consensus chain receives a fraud proof (which is acceptable until domain block `#266`) but can not slash the malicious operator as it is already deregistered.

Reply:
> My issue with using domain blocks is that we would need to track every domains block height in the unlocks list then. Is that a valid concern?

I think it won't bring extra cost compared to the current approach, we just need to change the index of the unlock list from `consensus_block_number` to `(domain_id, domain_block_number)` unless I miss something.

Reply:
> I think it won't bring extra cost compared to the current approach, we just need to change the index of the unlock list from consensus_block_number to (domain_id, domain_block_number) unless I miss something.

Not just that, but we also need to adjust the other Storage items like PendingWithdrawals. This will be also be scoped such that each with withdrawal is on FIFO order. May not be such a trivial change since the underlying assumption has changed.

> StakeEpochDuration = 5

This is not the final number and has a TODO to revisit this number.

 Looking at the example there, I dont see that practically but this was the known concern during our offline discussion. 
I do not have a strong opinion here but we need to update spec accordingly once we make the decision since calling it Global pending unlocks is confusing as well.

The change will be handled as either a separate PR or part of the epoch PR that uses block tree.